File: backend/src/main/java/com/back/global/dto/PageResponseDto.java
package com.back.global.dto;

import org.springframework.data.domain.Page;

import java.util.List;

public record PageResponseDto<T>(
        List<T> data,
        int pageNumber,
        int pageSize,
        int totalPages,
        long totalElements,
        boolean isLast
) {
    public PageResponseDto(Page<T> page) {
        this(
                page.getContent(),
                page.getNumber(),
                page.getSize(),
                page.getTotalPages(),
                page.getTotalElements(),
                page.isLast()
        );
    }
}

File: backend/src/main/java/com/back/global/dto/PageResponseDto.kt
package com.back.global.dto

import org.springframework.data.domain.Page

data class PageResponseDto<T>(
    val data: MutableList<T>,
    val pageNumber: Int,
    val pageSize: Int,
    val totalPages: Int,
    val totalElements: Long,
    val isLast: Boolean
) {
    constructor(page: Page<T>) : this(
        page.getContent(),
        page.getNumber(),
        page.getSize(),
        page.getTotalPages(),
        page.getTotalElements(),
        page.isLast()
    )
}
